src/debugging/Inspector/scripts/generate.py

```diff
@@ -0,0 +1,197 @@
+        object_declarations = filter(lambda decl: isinstance(decl.type, ObjectType), domain.type_declarations)
+        enum_declarations = filter(lambda decl: isinstance(decl.type, EnumType), domain.type_declarations)
+
+        for declaration in primitive_declarations:
+            description = declaration.description if declaration.description else 'No description'  
+            lines.append('export type {} = {} // {}'.format(declaration.type_name, ts_name_for_primitive_type(domain, declaration.type.aliased_type), description))
+
+        lines.append('')
+
+        for declaration in object_declarations:
+            lines.append('export interface %s {' % declaration.type.raw_name())
+            
+            declaration_properties = filter(lambda member: isinstance(member.type, (AliasedType, ObjectType ,PrimitiveType)), declaration.type_members)            
+            for declaration_property in declaration_properties:
+                member_args = {
+                    'name': declaration_property.member_name,
+                    'type': ts_name_for_primitive_type(domain, declaration_property.type),
+                    'optional': '?' if declaration_property.is_optional else '',
+                    'description': declaration_property.description                    
+                }
+                lines.append('\t%(name)s%(optional)s: %(type)s; // %(description)s' % member_args)
+
+            lines.append('}\n')
+
+        for declaration in enum_declarations:
+            stylized_enums = map(stylized_name_for_enum_value, declaration.type.enum_values())
+            lines.append('export enum %s { %s }; \n' % (declaration.type.raw_name(), ', '.join(stylized_enums)))
+
+        return lines
+
+    def generate_domain_commands(self, domain):
+        lines = []
+        for command in domain.commands:
+            returnParams = ", ".join(['%s: %s' % (parameter.parameter_name, ts_name_for_primitive_type(domain, parameter.type)) for parameter in command.return_parameters])
+
+            command_args = {
+                'domain': domain.domain_name,
+                'commandName': command.command_name,
+                'callParams': ", ".join([self.generate_parameter_object(domain, parameter) for parameter in command.call_parameters]),
+                'returnParams': 'void' if len(returnParams) == 0 else '{ %s }' % returnParams,
+                'description': command.description if command.description else 'No description'
+            }
+            lines.append('\t%(commandName)s(%(callParams)s): %(returnParams)s; // %(description)s' % command_args)
+        return lines
+
+    def generate_domain_events(self, domain):
+        lines = []
+        for event in domain.events:            
+            argument_index = 0;
+            parameters = []
+            for parameter in event.event_parameters:
+                parameters.append('\"%s\": arguments[%s]' % (parameter.parameter_name, argument_index))
```

Comment:
I just wonder why did you went with using `arguments[i]` instead of the parameter name.